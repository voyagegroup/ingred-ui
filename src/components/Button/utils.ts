import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
