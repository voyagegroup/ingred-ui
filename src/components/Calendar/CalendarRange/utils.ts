import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
