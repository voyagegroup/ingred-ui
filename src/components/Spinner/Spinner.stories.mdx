import { Meta, Story, ArgsTable, Canvas } from "@storybook/addon-docs/blocks";
import Spinner from "./Spinner";

<Meta title="Components/Feedback/Spinner" component={Spinner} />

# Spinner

<ArgsTable of={Spinner} />

## Samples

<Canvas>
  <Story name="Example" args={{}}>
    {(args) => <Spinner {...args} />}
  </Story>
</Canvas>
