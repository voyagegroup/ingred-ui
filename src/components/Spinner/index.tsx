export { default, SpinnerProps } from "./Spinner";
