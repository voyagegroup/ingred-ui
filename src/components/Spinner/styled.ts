import styled, { keyframes } from "styled-components";
import { Property } from "csstype";

const rotate = keyframes`
  100% {
    transform: rotate(360deg);
  }
`;

const dash = keyframes`
  0% {
    stroke-dasharray: 1, 200;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -35px;
  }
  100% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -124px;
  }
`;

export const Container = styled.div<{ width: Property.Width }>`
  width: ${({ width }) => width};
  height: ${({ width }) => width};
  background-color: transparent;
`;

export const Svg = styled.svg`
  width: 100%;
  height: 100%;
  animation: ${rotate} 2s linear infinite;
  transform-origin: center center;
`;

export const Circle = styled.circle`
  stroke-width: 3;
  fill: none;
`;

export const PrimaryCircle = styled(Circle)`
  stroke-dasharray: 1, 200;
  stroke-dashoffset: 0;
  animation: ${dash} 1.5s ease-in-out infinite;
  stroke-linecap: round;
  stroke-miterlimit: 10;
`;
