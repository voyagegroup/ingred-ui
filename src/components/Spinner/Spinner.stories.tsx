import React from "react";
import { Story } from "@storybook/react/types-6-0";
import { Title, ArgsTable, Stories } from "@storybook/addon-docs";
import Spinner, { SpinnerProps } from "./Spinner";

export default {
  title: "Components/Feedback/Spinner",
  component: Spinner,
  parameters: {
    docs: {
      source: { type: "code" },
      page: () => (
        <>
          <Title />
          <ArgsTable of={Spinner} />
          <Stories includePrimary title="Stories" />
        </>
      ),
    },
  },
};

export const Example: Story<SpinnerProps> = (args) => <Spinner {...args} />;
