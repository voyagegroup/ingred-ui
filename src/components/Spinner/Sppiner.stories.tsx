import * as React from "react";
import Spinner from "./Spinner";

export default {
  title: "Components/Spinner",
  component: Spinner,
  parameters: {
    docs: { page: null },
  },
};

export const Overview = () => (
  <div>
    <Spinner />
  </div>
);
