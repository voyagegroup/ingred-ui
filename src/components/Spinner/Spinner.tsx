import * as React from "react";
import { Property } from "csstype";
import * as Styled from "./styled";
import { useTheme } from "../../themes/useTheme";

export type SpinnerProps = {
  width?: Property.Width;
};

const Spinner = React.forwardRef<HTMLDivElement, SpinnerProps>(function Spinner(
  { width = "50px" },
  ref,
) {
  const theme = useTheme();
  return (
    <Styled.Container ref={ref} width={width}>
      <Styled.Svg viewBox="25 25 50 50">
        <Styled.Circle
          cx="50"
          cy="50"
          r="20"
          stroke={theme.palette.gray.light}
        />
        <Styled.PrimaryCircle
          cx="50"
          cy="50"
          r="20"
          stroke={theme.palette.primary.main}
        />
      </Styled.Svg>
    </Styled.Container>
  );
});

export default Spinner;
