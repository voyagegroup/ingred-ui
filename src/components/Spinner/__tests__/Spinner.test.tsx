import * as React from "react";
import "@testing-library/jest-dom";
import { cleanup } from "@testing-library/react";
import Spinner from "../";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

describe("Spinner component testing", () => {
  afterEach(cleanup);

  test("Spinner", () => {
    const { asFragment } = renderWithThemeProvider(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
