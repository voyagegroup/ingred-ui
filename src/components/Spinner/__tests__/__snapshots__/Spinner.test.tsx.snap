// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component testing Spinner 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL dwAwYF"
    width="50px"
  >
    <svg
      class="sc-gsDKAQ htNwEJ"
      viewBox="25 25 50 50"
    >
      <circle
        class="sc-dkPtRN iJLnej"
        cx="50"
        cy="50"
        r="20"
        stroke="#E2E8EA"
      />
      <circle
        class="sc-dkPtRN sc-hKwDye iJLnej CFLCo"
        cx="50"
        cy="50"
        r="20"
        stroke="#0B82F4"
      />
    </svg>
  </div>
</DocumentFragment>
`;
